Userland/Libraries/LibGL/math/mat4.h

```diff
@@ -0,0 +1,244 @@
+        ret.m_data[2][2] = m_data[2][2] + rhs.m_data[2][2];
+        ret.m_data[2][3] = m_data[2][3] + rhs.m_data[2][3];
+
+        ret.m_data[3][0] = m_data[3][0] + rhs.m_data[3][0];
+        ret.m_data[3][1] = m_data[3][1] + rhs.m_data[3][1];
+        ret.m_data[3][2] = m_data[3][2] + rhs.m_data[3][2];
+        ret.m_data[3][3] = m_data[3][3] + rhs.m_data[3][3];
+
+        return ret;
+    }
+
+    Mat4& operator+=(const Mat4& rhs)
+    {
+        m_data[0][0] += rhs.m_data[0][0];
+        m_data[0][1] += rhs.m_data[0][1];
+        m_data[0][2] += rhs.m_data[0][2];
+        m_data[0][3] += rhs.m_data[0][3];
+
+        m_data[1][0] += rhs.m_data[1][0];
+        m_data[1][1] += rhs.m_data[1][1];
+        m_data[1][2] += rhs.m_data[1][2];
+        m_data[1][3] += rhs.m_data[1][3];
+
+        m_data[2][0] += rhs.m_data[2][0];
+        m_data[2][1] += rhs.m_data[2][1];
+        m_data[2][2] += rhs.m_data[2][2];
+        m_data[2][3] += rhs.m_data[2][3];
+
+        m_data[3][0] += rhs.m_data[3][0];
+        m_data[3][1] += rhs.m_data[3][1];
+        m_data[3][2] += rhs.m_data[3][2];
+        m_data[3][3] += rhs.m_data[3][3];
+
+        return *this;
+    }
+
+    Mat4 operator-(const Mat4& rhs)
+    {
+        Mat4 ret;
+
+
+        ret.m_data[0][0] = m_data[0][0] - rhs.m_data[0][0];
+        ret.m_data[0][1] = m_data[0][1] - rhs.m_data[0][1];
+        ret.m_data[0][2] = m_data[0][2] - rhs.m_data[0][2];
+        ret.m_data[0][3] = m_data[0][3] - rhs.m_data[0][3];
+
+        ret.m_data[1][0] = m_data[1][0] - rhs.m_data[1][0];
+        ret.m_data[1][1] = m_data[1][1] - rhs.m_data[1][1];
+        ret.m_data[1][2] = m_data[1][2] - rhs.m_data[1][2];
+        ret.m_data[1][3] = m_data[1][3] - rhs.m_data[1][3];
+
+        ret.m_data[2][0] = m_data[2][0] - rhs.m_data[2][0];
+        ret.m_data[2][1] = m_data[2][1] - rhs.m_data[2][1];
+        ret.m_data[2][2] = m_data[2][2] - rhs.m_data[2][2];
+        ret.m_data[2][3] = m_data[2][3] - rhs.m_data[2][3];
+
+        ret.m_data[3][0] = m_data[3][0] - rhs.m_data[3][0];
+        ret.m_data[3][1] = m_data[3][1] - rhs.m_data[3][1];
+        ret.m_data[3][2] = m_data[3][2] - rhs.m_data[3][2];
+        ret.m_data[3][3] = m_data[3][3] - rhs.m_data[3][3];
+
+        return ret;
+    }
+
+    Mat4& operator-=(const Mat4& rhs)
+    {
+        m_data[0][0] -= rhs.m_data[0][0];
+        m_data[0][1] -= rhs.m_data[0][1];
+        m_data[0][2] -= rhs.m_data[0][2];
+        m_data[0][3] -= rhs.m_data[0][3];
+
+        m_data[1][0] -= rhs.m_data[1][0];
+        m_data[1][1] -= rhs.m_data[1][1];
+        m_data[1][2] -= rhs.m_data[1][2];
+        m_data[1][3] -= rhs.m_data[1][3];
+
+        m_data[2][0] -= rhs.m_data[2][0];
+        m_data[2][1] -= rhs.m_data[2][1];
+        m_data[2][2] -= rhs.m_data[2][2];
+        m_data[2][2] -= rhs.m_data[2][2];
+
+        m_data[3][0] -= rhs.m_data[3][0];
+        m_data[3][1] -= rhs.m_data[3][1];
+        m_data[3][2] -= rhs.m_data[3][2];
+        m_data[3][2] -= rhs.m_data[3][2];
```

Comment:
Can you please change this to a loop? If you really want to force it to be unrolled you can use a pragma to force the compiler to do so, but I think that having the code expanded like this makes it harder to read and unnecessarily harder to modify.

Reply:
Not only that... but the code you referenced contains a copy-paste bug (`[2]` is used twice)